### ISSUE TITLE
5.0.0 - Crash with prefer-const rule where code contains array destructuring assignment with 'ignored values'

### ISSUE THREAD

**Issue by mjmasn** [eslint/eslint#10520]
This is a minimal reproduction for a bug I've found in `eslint@5.0.0`. As far as I can tell the example code below is valid JavaScript (see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Ignoring_some_returned_values).

**Tell us about your environment**

* **ESLint Version:** 5.0.0
* **Node Version:** 8.11.0
* **npm Version:** 5.6.0
* **yarn Version:** 1.7.0

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

```json
{
    "parserOptions": {
        "ecmaVersion": 2018
    },
    "rules": {
      "prefer-const": 1
    }
}

```

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
const b = [1, 2];
let a;

[, a] = b;

```

```bash
cd eslint-5-array-destructuring-assignment-bug
yarn eslint .
```

**What did you expect to happen?**
eslint would lint my project

**What actually happened? Please include the actual, raw output from ESLint.**
eslint did not lint my project and threw a TypeError from one of its built-in rules (`prefer-const`)
```
Cannot read property 'name' of null
TypeError: Cannot read property 'name' of null
    at elements.map.element (/node_modules/eslint/lib/rules/prefer-const.js:173:49)
    at Array.map (<anonymous>)
    at getIdentifierIfShouldBeConst (/node_modules/eslint/lib/rules/prefer-const.js:173:26)
    at groupByDestructuring (/node_modules/eslint/lib/rules/prefer-const.js:227:28)
    at Program:exit (/node_modules/eslint/lib/rules/prefer-const.js:362:17)
    at listeners.(anonymous function).forEach.listener (/node_modules/eslint/lib/util/safe-emitter.js:45:58)
    at Array.forEach (<anonymous>)
    at Object.emit (/node_modules/eslint/lib/util/safe-emitter.js:45:38)
    at NodeEventGenerator.applySelector (/node_modules/eslint/lib/util/node-event-generator.js:251:26)
    at NodeEventGenerator.applySelectors (/node_modules/eslint/lib/util/node-event-generator.js:280:22)
```

**Comment by platinumazure** [comment]
Hi @mjmasn, thanks for the issue!

**Comment by mjmasn** [comment]
:+1: BTW changing https://github.com/eslint/eslint/blob/196c1021c070e52259f2ceccc4f141bdb7456c60/lib/rules/prefer-const.js#L173 to
```js
.map(element => element ? element.name : null)
```
fixes the crash but I'm not sure if there are any wider implications...

**Comment by platinumazure** [comment]
Thanks @mjmasn, that makes sense. Would you like to create a PR (maybe with a new test that would trigger the crash)?

**Comment by platinumazure** [comment]
Just out of curiosity, anyone know if this can be recreated in the 4.x release line?

**Comment by mjmasn** [comment]
@platinumazure I don't think so, the code was introduced in https://github.com/eslint/eslint/commit/240c1a4df321de56c12acbcfad21734944d7e116 (first released in [5.0.0-alpha.4](https://github.com/eslint/eslint/releases/tag/v5.0.0-alpha.4))

**Comment by Aladdin-ADD** [comment]
should we release a patch `5.0.2`? cc @eslint/eslint-team
